modules/compiler/utils/source/barrier_regions.cpp

```diff
@@ -855,9 +892,16 @@ void compiler::utils::Barrier::MakeLiveVariableMemType() {
   const auto &dl = module_.getDataLayout();
 
   struct member_info {
+    /// @brief The root `value` being stored.
     Value *value;
+    /// @brief The member index of this member inside `value`, if `value` is a
+    /// decomposed structure type. Zero otherwise.
```

Comment:
is zero not a valid index or did you just want to come up with a good default value (although these values don't have defaults here).

Reply:
Zero is intended to be valid but it's rather arbitrary. It could also be `std::nullopt` but then we'd have to treat the regular and decomposed paths separately. By unifying the two code paths into iterating over a vector of types (the default case being a vector of one type), zero comes out naturally.

I was also kind of thinking of it like a representation of the address offset of the variable inside its parent (or itself), as in a struct. The pointer to the struct is the same address as the pointer to its first element.